elpis/engines/common/input/clean_json.py

```diff
@@ -103,57 +100,82 @@ def is_valid_utterance(clean_words: List[str],
 
     # Exclude utterance if langid thinks its english
     if remove_english and use_langid:
+        langid_identifier = LanguageIdentifier.from_modelstring(model, norm_probs=True)
         lang, prob = langid_identifier.classify(cleaned_transcription)
         if lang == "en" and prob > 0.5:
             return False
     return True
 
 
-def clean_json_data(json_data: List[Dict[str, str]],
-                    remove_english: bool = False,
-                    use_langid: bool = False,
-                    punctuation_to_collapse_by: str = '',
-                    punctuation_to_explode_by: str = '') -> List[Dict[str, str]]:
+def clean_json_utterance(utterance: Dict[str, str],
+                         remove_english: bool = False,
+                         use_langid: bool = False,
+                         punctuation_to_collapse_by: str = '',
+                         punctuation_to_explode_by: str = '') -> Dict[str, str]:
     """
-    Clean a list of utterances (Python dictionaries) based on the given parameters.
-    :param json_data: list of Python dictionaries, each must have a 'transcription' key-value.
+    Clean an utterance (Python dictionary) based on the given parameters.
+    :param utterance: Python dictionary, must have a 'transcript' key-value.
     :param remove_english: whether or not to remove English from the utterances.
     :param use_langid: whether or not to use the langid library to identify English to remove.
     :param punctuation_to_collapse_by: punctuation marks to strip
     :param punctuation_to_explode_by: punctuation marks to replace with spaces
-    :return: cleaned list of utterances (list of dictionaries).
+    :return: cleaned utterances (dictionary).
```

Comment:
If you're really changing this to deal with a single utterance, this will have to change, too.

Reply:
ok, done